src/workerd/api/r2-bucket.h

```diff
@@ -98,6 +100,20 @@ class R2Bucket: public jsg::Object {
     JSG_STRUCT(onlyIf, httpMetadata, customMetadata, md5, sha1, sha256, sha384, sha512);
   };
 
+  struct MultipartOptions {
+    jsg::Optional<kj::OneOf<HttpMetadata, jsg::Ref<Headers>>> httpMetadata;
+    jsg::Optional<jsg::Dict<kj::String>> customMetadata;
+
+    JSG_STRUCT(httpMetadata, customMetadata);
+  };
+
+  struct UploadedPart {
```

Comment:
This should be in r2-multipart.h I suspect.

Reply:
I don't think I can make that work, since I need to forward-declare anything I want to use from r2-multipart.h in r2-bucket.h. So I wouldn't be able to properly use the UploadedPart struct here, which I need because the UploadPart implementation is in r2-bucket as well.

Reply:
I can make that work if i move the implementation though, I'll get on that :)